lib/galaxy/tool_util/parser/cwl.py

```diff
@@ -158,7 +253,23 @@ def parse_requirements_and_containers(self):
         ))
 
     def parse_profile(self):
-        return "16.04"
+        return "17.09"
+
+    def parse_xrefs(self):
+        return []
+
+    def parse_provided_metadata_style(self):
+        return "default"
+
+    def parse_cores_min(self):
+        for h in self.tool_proxy.hints_or_requirements_of_class("ResourceRequirement"):
+            cores_min = h.get("coresMin")
+            if cores_min:
+                cores_min = int(cores_min)
+                if cores_min < 1:
+                    raise ValueError("cores_min must be a positive integer")
```

Comment:
This can be dynamic in CWL, e.g. in `dynresreq.cwl` :

```yaml
requirements:
  ResourceRequirement:
      ramMin: 8
      coresMin: $(inputs.special_file.size)
      coresMax: $(inputs.special_file.size)

inputs:
  special_file: File
```